lib/cucumber/core/compiler.rb

```diff
@@ -68,6 +71,10 @@ def source_lines_for_pickle(pickle)
         pickle.ast_node_ids.map { |id| source_line(id) }
       end
 
+      def source_lines_for_pickle_parents(pickle)
+        gherkin_query.scenario_parent_locations(pickle.ast_node_ids[0]).map(&:line)
```

Comment:
Just briefly analysing this, is there a reason we're using index 0 here. I've not dived into the code enough. If you can paste something in here to explain this it'll speed it up for me.

Reply:
@luke-hill Great question, took me a minute to figure out why. So the `pickle` variable represents a scenario, and _usually_ a scenario only has one corresponding line in the document... the line that starts with `Scenario:`, so we would expect the accessor to actually be `#ast_node_id`. However, there are circumstances in which a Scenario can have multiple lines, for example:

```
Scenario Outline: eating
  Given there are <start> cucumbers
  When I eat <eat> cucumbers
  Then I should have <left> cucumbers

  Examples:
    | start | eat | left |
    |    12 |   5 |    7 |
    |    20 |   5 |   15 |
```
In this case, the scenario has _two_ lines, the line beginning with `Scenario Outline:` and then also the line containing the values, e.g. `| 12 | 5 | 7 |`. Therefore, the pickle has to have an Array accessor of `#ast_node_ids`.

However, in our use-case, finding the parent lines of a given scenario, we don't care which scenario line is used, because they both have the same parents, and there's always at least one, so we can just grab the first one with `[0]`.

That's a lot of explanation! We should probably put this in the code somewhere? Maybe extract a well-named variable?

```  
      def source_lines_for_pickle_parents(pickle)
        first_scenario_line_ast_node_id = pickle.ast_node_ids.first # Scenario Outlines have multiple lines per scenario, but they both have the same parents
        gherkin_query.scenario_parent_locations(first_scenario_line_ast_node_id).map(&:line)
      end
```

WDYT?

Reply:
@luke-hill Taking another look at this, if we change `#matching_locations` to a `Set`, rather than an `Array`, we could just `pickle.ast_node_ids.flat_map(&:line)` and call it a day. Set semantics would get rid of the duplicates. Which, now that I think about it, honestly wouldn't really be an issue, if we just left it as an array. Who cares if there's duplicate lines? The array just needs to match the given line number or not.

Reply:
Given this is an ancient piece of buggy functionality, I'm happy to go with whatever is easier.

But I'm trying to nuke huge 100's / 1000's of rubocop offenses. So if this is something you think would be better placed dealt with easily now and then refactored later I'm happy if you think we'll get to it later.